---
title: "Lab 8"
author: "Chris Colvin"
date: "11/16/2020"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

# Exercise 1
#NGPhylogeny.fr is a web server that helps one to see the relationships of small set sequences. The server helped with sequence alignment along with phylogenetic analysis through the visualization of data in a phylogenetic tree. 

```{r}
ppi <- 300
png("tree_00.png", width=3*ppi, height=3*ppi, res=ppi)
```


```{r}
if (!requireNamespace("BiocManager", quietly = TRUE))
    install.packages("BiocManager")

BiocManager::install("ggtree")
BiocManager::install("treeio")
library(tidyverse)
library(ggtree)
library(treeio)
browseVignettes("ggtree")
```

```{r}
library(tidyverse)
library(ggtree)
# added treeio and ggimage library
library(treeio)
library(ggimage)
# need to install libmagick 
#sudo apt-get install libmagick++-dev
```
```{r}
library(tidyverse)
library(ggtree)
tree <- read.tree("data/tree_newick.nwk")
tree
```
```{r}
# build a ggplot with a geom_tree
ggplot(tree) + geom_tree() + theme_tree()
# This is convenient shorthand
ggtree(tree)
```

```{r}
# add a scale
ggtree(tree) + geom_treescale()
# or add the entire scale to the x axis with theme_tree2()
ggtree(tree) + theme_tree2()
```
```{r}
ggtree(tree, branch.length="none")
```

```{r}
ggtree(tree, branch.length="none", color="blue", size=2, linetype=3)

```
# Exercise 1

```{r}
ggtree(tree, layout='slanted')
ggtree(tree,layout='circular')
ggtree(tree,layout='circular', branch.length="none", color="red",size=3)
```
```{r}
# create the basic plot
p <- ggtree(tree)
# add node points
p + geom_nodepoint()
# add tip points
p + geom_tippoint()
# Label the tips
p + geom_tiplab()
```

# Exercise 2
```{r}
ggtree(tree) + geom_tiplab(color="purple") + geom_tippoint(color="purple") + geom_nodepoint(color="yellow", alpha = 5, size = 4) + ggtitle("Exercise 2 Phylogeny")
```

```{r}
ggtree(tree) + geom_text(aes(label=node), hjust=-.3)
```

```{r}
ggtree(tree) + geom_tiplab()

```

```{r}
MRCA(tree, c("C", "E"))
MRCA(tree, c("G", "H"))

```

```{r}
ggtree(tree) + 
  geom_cladelabel(node=17, label="Some random clade", color="red")
ggtree(tree) + 
  geom_tiplab() + 
  geom_cladelabel(node=17, label="Some random clade", 
                  color="red2", offset=.8)
ggtree(tree) + 
  geom_tiplab() + 
  geom_cladelabel(node=17, label="Some random clade", 
                  color="red2", offset=.8) + 
  geom_cladelabel(node=21, label="A different clade", 
                  color="blue", offset=.8)
```

```{r}
ggtree(tree) + 
  geom_tiplab() + 
  geom_cladelabel(node=17, label="Some random clade", 
                  color="red2", offset=.8, align=TRUE) + 
  geom_cladelabel(node=21, label="A different clade", 
                  color="blue", offset=.8, align=TRUE) + 
  theme_tree2() + 
  xlim(0, 70) + 
  theme_tree()
```

```{r}
ggtree(tree) + 
  geom_tiplab() + 
  geom_hilight(node=17, fill="gold") + 
  geom_hilight(node=21, fill="purple")
```

```{r}
ggtree(tree) + 
  geom_tiplab() + 
  geom_taxalink("E", "H", color="blue3") +
  geom_taxalink("C", "G", color="orange2", curvature=-.9)
```


```{r}
ggtree(tree) + theme_tree2() + geom_tiplab() + ggtitle("Exercise 3") + geom_hilight(node=19, fill="gold") + geom_hilight(node=23, fill="blue") + geom_taxalink(3,5, curvature=.5, linetype=2) + geom_taxalink(7,10, curvature=.5, linetype=2) + geom_cladelabel(node=17, label="Superclade 17", 
                  color="red2", offset=-7, align=TRUE)
```

```{r}
# Read the data
tree <- read.beast("data/flu_tree_beast.tree")
# supply a most recent sampling date so you get the dates
# and add a scale bar
ggtree(tree, mrsd="2013-01-01") + 
  theme_tree2() 
# Finally, add tip labels and adjust axis
ggtree(tree, mrsd="2013-01-01") + 
  theme_tree2() + 
  geom_tiplab(align=TRUE, linesize=.5) + 
  xlim(1990, 2020)
```

```{r}
msaplot(p=ggtree(tree), fasta="data/flu_aasequence.fasta", window=c(150, 175))
```


```{r}
mytree <- read.tree("fastree_result.tre")
ggtree(mytree, layout='circular') + xlim(0,3) + geom_tiplab(hjust=-.1, size = 3) +   geom_hilight(node=34, fill="gold") + geom_hilight(node=29, fill="blue") 
```

